### DISCUSSION TITLE
Move from OpenAPI Plugin to Swaggerhub

### DISCUSSION THREAD

**Discussion by maximilianong** [eclipse-tractusx/eclipse-tractusx.github.io#592] [Q&A] [answered]
For the docusaurus updates we want to get rid of the actual open API plugin that we are using: https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/issues/130

BPDM KIT especially @SujitMBRDI moved from this plugin to Swaggerhub.

He sees the following issue:

> "But I am seeing issue with it, Let’s consider I am creating new kit for Certificate Management Application.
> And now, I want to provide link of swagger hub to the Development view. Below is my swagger hub link for newly created application api’s.
> 
> https://app.swaggerhub.com/apis/**SujitMBRDI/**Certificate_Management_Api/0.0.1-SNAPSHOT
> 
> Here, I could not proceed without providing my GitHub or I need to create new GitHub user for application and create new repository for documents.
> 
> Kindly suggest me the way I should follow or confirm if it is fine to use personal git hub account."

@SebastianBezold / @Siegfriedk do you have an opinion on that?

**Comment by Siegfriedk** [comment, marked as answer]
@maximilianong we do have a swaggerhub access key in our tractusx org. He should upload it through a github action.

**Comment by SebastianBezold** [comment]
Hi @maximilianong and @SujitMBRDI,

like @Siegfriedk mentioned, we should publish our openAPI spec via `eclipse-tractusx` and not via personal accounts. 
This is also questionable, since copyright is most likely with your company and not you personally.

There are already examples on how to achieve this via GitHub workflow. See for example:

- [Digital Product Pass workflow](https://github.com/eclipse-tractusx/digital-product-pass/blob/main/.github/workflows/publish-swagger-hub.yaml)
- [EDC workflow](https://github.com/eclipse-tractusx/tractusx-edc/blob/main/.github/workflows/publish-swaggerhub.yaml)

It is always a good option to ask on our [Matrix Chat](https://matrix.to/#/#tractusx:matrix.eclipse.org) or the [Mailing List](https://accounts.eclipse.org/mailing-list/tractusx-dev) to get in contact with people who already achieved this
